### DISCUSSION TITLE
Please help me.I want to compare changes between two local BIM file versions. How can I compare the differences and obtain the changeset after the changes。

### DISCUSSION THREAD

**Discussion by MYDJH** [iTwin/itwinjs-core#6601] [Q&A] [answered]
"I want to read local BIM files, compare changes between two BIM files, and obtain the changeset. I found that the examples provided by iTwin involve using REST API for comparison to retrieve the changeset. I want to perform local comparisons. How can I achieve this?"
Here is the example address for using REST API to obtain the difference between two BIM files:[https://www.itwinjs.org/sandboxes/iTwinPlatform/Changed%20Elements](url)

I appreciate any insights, or guidance Thank you very much in advance!

**Comment by pmconne** [comment]
The change history of an iModel is stored as a series of changesets in iModelHub. If you're not using iModelHub, you won't be able to use changesets or version comparison. You can't take two local .bim files and produce a useful "diff" of them.

Are you using iModelHub?

**Reply by MYDJH** [reply]
"First of all, thank you very much for your response. It seems that my program doesn't use iModelHub to manage iModels. As for iTwin, I am a beginner. I obtained .bim files through iTwin Snapshot conversion, and then the program reads .bim files from the local environment to browse views. Now, I want to implement a functionality to compare changes between two .bim files. I'm unable to use the Changed Elements API via 
`'https://api.bentley.com/changedelements/comparisoniModelId=${iModelId}&iTwinId=${iTwinId}&startChangesetId=${startChangesetId}&endChangesetId=${endChangesetId}'` to obtain the changeset after comparison. I want to achieve local comparison between two local .bim files to obtain data in a format similar to changeset. Do you have any good suggestions?"

**Reply by pmconne** [reply, marked as answer]
Per [the docs](https://www.itwinjs.org/reference/core-backend/imodels/snapshotdb/):

> A snapshot iModel database file that is used for archival and data transfer purposes.

Like I said, there is no way to produce a useful "diff" between two arbitrary .bim files. The whole purpose of iModelHub is to track the change history of an iModel. That history is what permits comparisons between versions of the iModel at different points in time. A snapshot has no change history, therefore it is useless for collaboration workflows - hence the name "snapshot", meaning an immutable picture of the iModel at a specific point in time.

**Reply by MYDJH** [reply]
"I understand now. Thank you very much for your helpful response."
